### PR TITLE
WIP: LibRequest+LibWeb: Propagate HTTP status messages

### PR DESCRIPTION
**Pull request by yyny**
This fixes WPT test [wpt/cors/status-async](http://wpt.live/cors/status-async.htm)

### CODE REVIEW

File: Userland/Libraries/LibRequests/RequestClient.h [hunk 1]
```diff
@@ -42,7 +43,7 @@ class RequestClient final
     virtual void request_started(i32, IPC::File const&) override;
     virtual void request_finished(i32, u64, Optional<NetworkError> const&) override;
     virtual void certificate_requested(i32) override;
-    virtual void headers_became_available(i32, HTTP::HeaderMap const&, Optional<u32> const&) override;
+    virtual void headers_became_available(i32, u16, ByteBuffer const&, HTTP::HeaderMap const&) override;
```
Comment (alimpfard) [inline Userland/Libraries/LibRequests/RequestClient.h:46]: Please give these new parameters a name, it's too ambiguous as-is.

File: Userland/Libraries/LibHTTP/HttpStatus.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+#pragma once
+
+#include <AK/ByteString.h>
+#include <AK/Format.h>
+
+namespace HTTP {
+
+struct HttpStatus {
+    u16 code { 0 };
+    ByteBuffer reason_phrase;
```
Comment (alimpfard) [inline Userland/Libraries/LibHTTP/HttpStatus.h:16]: Yuck but okay.

**Review by alimpfard — COMMENTED** [review]
Looks vaguely okay, not exactly in love with the fact that wpt requires the server's response phrase to be relayed...

**Comment by rmg-x** [comment]
Dupe of #1876

**Pull request closed without merging**